music21/midi/__init__.py

```diff
@@ -1445,10 +1445,7 @@ def getChannels(self):
         >>> mt.getChannels()
         [5, 14]
         '''
-        post = []
-        for e in self.events:
-            if e.channel not in post:
-                post.append(e.channel)
+        post = { e.channel for e in self.events if e.channel is not None }
```

Comment:
Can you rewrite using the original code just changing `if e.channel not in post and e.channel is not None` ?  Or define a set at the beginning and just put `if e.channel is None: \n continue` before a `post.add(e.channel)` line.  There are a lot of beginner programmers who read music21, and in teaching them, I find that explicit iteration is much easier for them to grasp than filtered set comprehensions.

Otherwise happy to take after tests pass.

Reply:
Ah yes! I'll update the code to keep the same style